### PR TITLE
Normalize App parameter declaration (doc-include folders, boot)

### LINKED ISSUE #3039: Convention - Maybe remove & from $a (App) parameter?
I would like to bring up the issue as show here:
https://github.com/friendica/friendica/pull/3036#issuecomment-268282396

The & makes such thing possible:
```
function bla (App &$a) {
$a = new App();
}
```
Now outside code does not notice that $a has been changed (somewhere). It is currently not the case (as my trustworthy Midnight Commander has found nothing about it) but accidents like this should be prevented in the future (more code/features being added).

### PR DESCRIPTION
**Pull request by Hypolite**
Fixes #3039

This is a simple cosmetics change, removing redundant `&` operators, adding `App` type hints, a few spaces in function declaration lines, but nothing more.

I won’t accept any change suggestions (like standards or docs) not directly related with the relevant function declaration lines. The diff was so big that I had to split the changes between 4 other PRs:
- #3075 (view folder)
- #3077 (mod folder, 1 out of 3)
- #3078 (mod folder, 2 out of 3)
- #3079 (mod folder, 3 out of 3)

### CODE REVIEW

**Comment by Quix0r** [comment]
Fuck, didn't notice it was the same PR. :-(

File: doc/autoloader.md [hunk 1]
```diff
@@ -161,10 +161,10 @@ But if you want to use classes from another library, you need to use the full na
 ```
     <?php
     namespace \Frienidca;
```
Comment (annando) [inline doc/autoloader.md]: Who is "Frienidca"? :-)

File: doc/autoloader.md [hunk 2]
```diff
@@ -174,12 +174,12 @@ if you use that class in many places of the code and you don't want to write the
...
```
Comment (annando) [inline doc/autoloader.md]: Copy&Paste :-)

File: doc/autoloader.md (hunk 1 above)
Comment (Hypolite) [inline doc/autoloader.md]: You should ask @fabrixxm 😙 🎵

Comment (fabrixxm) [inline doc/autoloader.md]: gh...

**Pull request merged**